src/csharp/Grpc.Core/Logging/LogLevelFilterLogger.cs

```diff
@@ -27,16 +27,31 @@ namespace Grpc.Core.Logging
     /// <summary>Logger that filters out messages below certain log level.</summary>
     public class LogLevelFilterLogger : ILogger
     {
+        // Verbosity environment variable used by C core.
+        private const string CoreVerbosityEnvVarName = "GRPC_VERBOSITY";
         readonly ILogger innerLogger;
         readonly LogLevel logLevel;
 
         /// <summary>
         /// Creates and instance of <c>LogLevelFilter.</c>
         /// </summary>
-        public LogLevelFilterLogger(ILogger logger, LogLevel logLevel)
+        public LogLevelFilterLogger(ILogger logger, LogLevel logLevel) : this(logger, logLevel, false)
+        {
+        }
+
+        /// <summary>
+        /// Creates and instance of <c>LogLevelFilter.</c>
+        /// The <c>fromEnvironmentVariable</c> parameter allows looking up "GRPC_VERBOSITY" setting provided by C-core
+        /// and uses the same log level for C# logs. Using this setting is recommended as it makes the otherwise separate
+        /// C# and C-core logging settings work in lockstep and more intutively.
+        /// </summary>
+        /// <param name="logger">the logger to forward filtered logs to.</param>
+        /// <param name="defaultLogLevel">the default log level, unless overriden by env variable.</param>
+        /// <param name="fromEnvironmentVariable">if <c>true</c>, override log level with setting from environment variable.</param>
+        public LogLevelFilterLogger(ILogger logger, LogLevel defaultLogLevel, bool fromEnvironmentVariable)
```

Comment:
It feels like it would make more sense for this to delegate to the constructor above, that could just set the fields directly. There's no point in calling GetLogLevelFromEnvironment with a constant value of false for fromEnvironmentVariable.